src/sage/rings/polynomial/polynomial_element.pyx

```diff
@@ -2103,8 +2103,60 @@ cdef class Polynomial(CommutativePolynomial):
             sage: r = f.any_root()
             sage: r^2 + r
             a^2 + a
-        """
-        if self.base_ring().is_finite() and self.base_ring().is_field():
+
+        Check that issue 33329 is resolved: implementation used to loop infinitely
```

Comment:
We do, it is `:issue:` (see the `extlinks` variable in `sage/misc/sagedoc.py`).

Reply:
True... but it just works. Pull and issues have unique labelling so that both `https://github.com/sagemath/sage/issues/35326` and `https://github.com/sagemath/sage/pull/35326` point to the same page.